### ISSUE TITLE
Simplify API by providing fallback <slot> content

### ISSUE THREAD

**Issue by dlockhart** [BrightspaceUI/tile#23]
Currently to override the default `image-url` behaviour you need to set the `custom-image-format` attribute and also provide a `d2l-image-tile-image` slot.

Similarly, to use a menu you need to set the `show-menu` attribute and provide a `d2l-image-tile-menu` slot.

To simplify things, we should be able to use the shadow DOM concept of [fallback slot content](https://www.polymer-project.org/2.0/docs/devguide/shadow-dom#fallback-content).

For the custom image slot, we just put the default image behaviour inside the `<slot>`:

```html
<slot name="d2l-image-tile-image">
<div
	class="d2l-image-tile-image"
	role="presentation"
	style$="[[_getImageStyle(imgUrl)]]"></div>
</slot>
```

If they provide content for that `<slot>`, it'll get replaced with what they provide.

The menu may be slightly more complex, but possibly by coming up with a solution for #20 we could simplify things.

**Comment by jstefaniuk-d2l** [comment]
Taking this on (I can't assign myself)

**Comment by jstefaniuk-d2l** [comment]
Fixed via #26 #27 #38
